### ISSUE TITLE
Support using certificates for Azure Key vault. 

### ISSUE THREAD

**Issue by sp98** [libopenstorage/secrets#81]
**Is this a BUG REPORT or FEATURE REQUEST?**:
FEATURE REQUEST

**What happened**:
As of now, the azure key vault client can be created using the ClientID and ClientSecret.  The [recommended way ](https://learn.microsoft.com/en-us/azure/key-vault/general/developers-guide#authentication-best-practices)of creating the client is using a certificate. 

**What you expected to happen**:
Create azure key vault client using certificate. Use the `NewServicePrincipalTokenFromCertificate`  function. 

**How to reproduce it (as minimally and precisely as possible)**:
N/A

**Anything else we need to know?**:
N/A

**Environment**:
- Container Orchestrator and version:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by sp98** [comment]
@adityadani 

I'm from the [Rook](https://github.com/rook/rook) team. 
I would like to take this up and work on it. Just wanted to check if you are still accepting contributions.

**Comment by adityadani** [comment]
@sp98 Yes feel free to send out a PR and we will review it. cc @dahuang-purestorage
